dvc/repo/experiments/ls.py

```diff
@@ -3,49 +3,35 @@
 
 from dvc.repo import locked
 from dvc.repo.scm_context import scm_context
+from dvc.scm import last_n_commits
+from dvc.types import Optional
 
-from .utils import (
-    exp_refs,
-    exp_refs_by_baseline,
-    remote_exp_refs,
-    remote_exp_refs_by_baseline,
-)
+from .utils import exp_refs, exp_refs_by_baseline
 
 logger = logging.getLogger(__name__)
 
 
 @locked
 @scm_context
-def ls(repo, *args, rev=None, git_remote=None, all_=False, **kwargs):
-    from scmrepo.git import Git
-
-    from dvc.scm import RevError, resolve_rev
-
-    if rev:
-        try:
-            rev = resolve_rev(repo.scm, rev)
-        except RevError:
-            if not (git_remote and Git.is_sha(rev)):
-                # This could be a remote rev that has not been fetched yet
-                raise
-    elif not all_:
-        rev = repo.scm.get_rev()
-
+def ls(
+    repo,
+    rev: Optional[str] = None,
+    all_commits: bool = False,
+    num: int = 1,
+    git_remote: Optional[str] = None,
+):
     results = defaultdict(list)
-
-    if rev:
-        if git_remote:
-            gen = remote_exp_refs_by_baseline(repo.scm, git_remote, rev)
-        else:
-            gen = exp_refs_by_baseline(repo.scm, rev)
-        for info in gen:
-            results[rev].append(info.name)
-    elif all_:
-        if git_remote:
-            gen = remote_exp_refs(repo.scm, git_remote)
-        else:
-            gen = exp_refs(repo.scm)
+    if all_commits:
+        gen = exp_refs(repo.scm, git_remote)
```

Comment:
I'm reading this due to [discussion](https://github.com/iterative/dvc.org/pull/3191#discussion_r787060961) in iterative/dvc.org#3191 

`all_commits` seems to contain `refs/exps/` like experiment references, right?

Reply:
> I'm reading this due to [discussion](https://github.com/iterative/dvc.org/pull/3191#discussion_r787060961) in [iterative/dvc.org#3191](https://github.com/iterative/dvc.org/pull/3191)
> 
> `all_commits` seems to contain `refs/exps/` like experiment references, right?

`all_commits` are user commits here, while for `refs/exps/` they are generated by dvc as experiments although the storage format is `commits`.

Reply:
So it's the same commits as in `git log -n`?

Reply:
> So it's the same commits as in `git log -n`?

Yes, it is.

Reply:
Regarding `-n 0`, `-n -1`, etc., `git log -n 0` returns nothing, while `git log -n -1` and `git log -n -2` return all history. If we want consistency with `git log -n`, let's go with this behavior.

Reply:
One question on `-n` though, `git log` defaults to no value there, meaning full history. We default to `-n 0`. Is there a way to indicate full history or is that left for some other future update?

Reply:
Yes, like `git log -n -1` and `git log -n -2`, `dvc exp list -n -1` or `dvc exp list -n -2` should indicate full history.

Reply:
Not sure I follow. By "full history" I mean to include all ancestor commits, equivalent to `--all-commits` if there's a single branch in the repo.

I guess if you give a really large number like `-n -9999` that excedes the no. of ancestor commits that should do the trick.

Reply:
I think the suggestion was that `-n` with any negative number should give all ancestors of the specified revision the same way that it does in git (it doesn't matter if it's `-1` or `-99999`). `-A/--all-commits` is different since it applies to all branches in the repo.

Reply:
> I think the suggestion was that `-n` with any negative number should give all ancestors of the specified revision the same way that it does in git (it doesn't matter if it's `-1` or `-99999`). `-A/--all-commits` is different since it applies to all branches in the repo.

The current discussion on this is that we follow the results from `git log -n`, `0` for a empty list, a negative value for the whole history, positive value for a list with `n` members.